### PR TITLE
allow some of the istio and addon deployments to be deployed on openshift

### PR DESCRIPTION
**Pull request by jmazzitelli**
fixes: https://github.com/kiali/kiali/issues/6635

### CODE REVIEW

**Comment by jmazzitelli** [comment]
I installed the latest 1.20 dev build with this script and it works.

```
hack/istio/download-istio.sh -div 1.20-dev

hack/istio/install-istio-via-istioctl.sh -c oc -ih default -it default

$ oc get pods -n istio-system 
NAME                                    READY   STATUS    RESTARTS   AGE
grafana-7bd5db55c4-7z6q9                1/1     Running   0          5m11s
istio-egressgateway-69f7c4cf6d-nsjst    1/1     Running   0          5m20s
istio-ingressgateway-58b7b888bc-xzfzs   1/1     Running   0          5m20s
istiod-d58fc696-9fcc6                   1/1     Running   0          5m24s
jaeger-78756f7d48-gqnzq                 1/1     Running   0          5m11s
prometheus-67f6764db9-6bhv6             2/2     Running   0          5m12s
```

**Comment by jmazzitelli** [comment]
For some reason, `groups: ["system:serviceaccount:istio-system"]` was not taking effect, even though it should (see https://www.redhat.com/sysadmin/security-context-constraint-configuration). So that is why I added the additional `users` section in the new SCC the script creates. I didn't have time to debug that, so I just added prometheus and grafana SAs in the users section.

**Comment by jmazzitelli** [comment]
This doesn't work on OpenShift 4.12 - error results:
```
        message: 'container has runAsNonRoot and image will run as root (pod: "istio-ingressgateway-5c9f5959cb-t9hzd_istio-system(0768b3e0-22b3-4457-acc1-9a7f4f144889)"
```

yet on 4.13 it works fine.

**Comment by jmazzitelli** [comment]
If you run with istio 1.19 it works on openshift 4.12 (though you get a really odd stack trace that I can't explain coming from istioctl). So I think because this fails with a dev build of istio, we can ignore that for now.

Here's that stack trace, though:
```
Installing Istio...
✔ Istio core installed                                                                                                                     
✔ Istiod installed                                                                                                                         
  Processing resources for CNI, Egress gateways, Ingress gateways.                                                                         [controller-runtime] log.SetLogger(...) was never called, logs will not be displayed:
goroutine 25 [running]:
runtime/debug.Stack()
	runtime/debug/stack.go:24 +0x5e
sigs.k8s.io/controller-runtime/pkg/log.eventuallyFulfillRoot()
	sigs.k8s.io/controller-runtime@v0.15.1/pkg/log/log.go:59 +0xcd
sigs.k8s.io/controller-runtime/pkg/log.(*delegatingLogSink).Enabled(0xc0009984c0, 0x8c00000000000000?)
	sigs.k8s.io/controller-runtime@v0.15.1/pkg/log/deleg.go:111 +0x32
github.com/go-logr/logr.Logger.Enabled(...)
	github.com/go-logr/logr@v1.2.4/logr.go:261
github.com/go-logr/logr.Logger.Info({{0x35b93f8?, 0xc0009984c0?}, 0xc001242700?}, {0xc001242700, 0x304}, {0x0, 0x0, 0x0})
	github.com/go-logr/logr@v1.2.4/logr.go:274 +0x72
sigs.k8s.io/controller-runtime/pkg/log.(*KubeAPIWarningLogger).HandleWarningHeader(0xc0008f29c0?, 0x1?, {0x3013bf4?, 0x7?}, {0xc001242700?, 0x304?})
	sigs.k8s.io/controller-runtime@v0.15.1/pkg/log/warning_handler.go:65 +0x167
k8s.io/client-go/rest.handleWarnings(0xc001aac4b0?, {0x35795c0?, 0xc0008f29c0?})
	k8s.io/client-go@v0.28.0/rest/warnings.go:144 +0xef
k8s.io/client-go/rest.(*Request).transformResponse(0xc001aaafc0, 0xc0019b78c0, 0xc001c18f00)
	k8s.io/client-go@v0.28.0/rest/request.go:1179 +0x5c8
k8s.io/client-go/rest.(*Request).Do.func1(0xc00117ba40?, 0x0?)
	k8s.io/client-go@v0.28.0/rest/request.go:1064 +0x31
k8s.io/client-go/rest.(*Request).request.func3.1(...)
	k8s.io/client-go@v0.28.0/rest/request.go:1039
k8s.io/client-go/rest.(*Request).request.func3(0xc0019b78c0, 0xc0013cb170, {0x35b1e18?, 0xc00117ba40?}, 0x0?, 0x0?, 0xc000b884a0?, {0x0?, 0x0?}, 0x32a87f0)
	k8s.io/client-go@v0.28.0/rest/request.go:1046 +0xd7
k8s.io/client-go/rest.(*Request).request(0xc001aaafc0, {0x35b1c58, 0x53c38a0}, 0x28a8460?)
	k8s.io/client-go@v0.28.0/rest/request.go:1048 +0x4ed
k8s.io/client-go/rest.(*Request).Do(0xc001aaafc0, {0x35b1c58, 0x53c38a0})
	k8s.io/client-go@v0.28.0/rest/request.go:1063 +0xb0
sigs.k8s.io/controller-runtime/pkg/client.(*unstructuredClient).Patch(0xc000a3c0a8, {0x35b1c58, 0x53c38a0}, {0x35e2230?, 0xc001826390?}, {0x3599258, 0x53c38a0}, {0xc000b88380, 0x2, 0x2})
	sigs.k8s.io/controller-runtime@v0.15.1/pkg/client/unstructured_client.go:165 +0x4be
sigs.k8s.io/controller-runtime/pkg/client.(*client).Patch(0xc000a3c090, {0x35b1c58, 0x53c38a0}, {0x35e2230?, 0xc001826390}, {0x3599258, 0x53c38a0}, {0xc000b88380, 0x2, 0x2})
	sigs.k8s.io/controller-runtime@v0.15.1/pkg/client/client.go:332 +0x365
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).serverSideApply(0xc000cfbae0, 0xc001826390)
	istio.io/istio/operator/pkg/helmreconciler/apply.go:237 +0x305
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).ApplyObject(0xc000cfbae0, 0xc001826390, 0x1)
	istio.io/istio/operator/pkg/helmreconciler/apply.go:184 +0x545
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).ApplyManifest(0xc000cfbae0, {{0x300e671?, 0x3?}, {0xc000fa6000?, 0x1f34?}}, 0x1?)
	istio.io/istio/operator/pkg/helmreconciler/apply.go:102 +0x1566
istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).processRecursive.func1()
	istio.io/istio/operator/pkg/helmreconciler/reconciler.go:223 +0x270
created by istio.io/istio/operator/pkg/helmreconciler.(*HelmReconciler).processRecursive in goroutine 1
	istio.io/istio/operator/pkg/helmreconciler/reconciler.go:200 +0xbf
✔ Egress gateways installed                                                                                                                
✔ Ingress gateways installed                                                                                                               
✔ CNI installed                                                                                                                            
✔ Installation complete                                                                                                                    
```

**Review by jshaughn — APPROVED** [review]
OK, so the script is fine running with Istio 1.19 on CRC 4.12.  But it is not fine running against 1.20 dev build.  In that case pods ingress/egress pods won't start because they are trying to start as root.  That may just be an issue with the dev build.  So, I'll approve, but we need to vet 1.20 at a later point.

File: hack/istio/install-istio-via-istioctl.sh [hunk 1]
```diff
@@ -330,13 +341,35 @@ if [ "${DELETE_ISTIO}" != "true" ]; then
...
+supplementalGroups:
+  type: RunAsAny
+fsGroup:
+  type: RunAsAny
+allowPrivilegedContainer: true
+seccompProfiles:
+- '*'
+priority: 9
+groups:
+- "system:serviceaccount:${NAMESPACE}"
```
Comment (FilipB) [inline hack/istio/install-istio-via-istioctl.sh:368]: This means that `allowPrivilegedContainer: true`  will be applied to all SAs in istio-system. Shouldn't we add that just for prometheus and grafana SAs? Other SA should follow OCP platform setup specified in Istio doc - `adm policy add-scc-to-group anyuid system:serviceaccounts:${NAMESPACE}`

Comment (jmazzitelli) [reply hack/istio/install-istio-via-istioctl.sh:368]: https://github.com/kiali/kiali/pull/6640

**Pull request merged**